packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -63,6 +68,15 @@ export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
       },
     )
 
+  const sessionHistory = useMemo(
+    () =>
+      sessionHistoryInfiniteData?.pages
+        .flatMap(page => page.page)
+        /* Note: session history is ordered from most recent to least recent in each page, so reverse the order when restoring the chat interface */
+        .reverse() ?? [],
+    [sessionHistoryInfiniteData],
+  )
```

Comment:
Could have just replaced the `reverse` with the new [Array.prototype.toReversed](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toReversed) function, but to support multiple pages you would have needed to add this useMemo, so might as well just call `reverse` here.